### PR TITLE
Include prereleases in semver check

### PR DESCRIPTION
**Pull request by beyang**
this apparently doesn't work on rc versions like `3.9.0-rc.2`

### CODE REVIEW

File: web/src/regression/util/init.ts [hunk 1]
```diff
@@ -26,15 +26,15 @@ const supportedSourcegraphVersionConstraint = '>=3.8'
  */
 export async function setTestDefaults(driver: Driver, gqlClient: GraphQLClient): Promise<void> {
     const version = await currentProductVersion(gqlClient)
```
Comment (lguychard) [inline web/src/regression/util/init.ts]:
```suggestion
    // const version = await currentProductVersion(gqlClient)
```

**Comment by felixfbecker** [comment]
My feeling is that there must be an easy proper fix for this.

**Comment by felixfbecker** [comment]
From the docs https://www.npmjs.com/package/semver#functions:

> `includePrerelease` Set to suppress the default behavior of excluding prerelease tagged versions from ranges unless they are explicitly opted into.

Tried out in the REPL:

```js
> semver.satisfies('3.9.0-rc.2', '>=3.8', { includePrerelease: true })
true
```

File: web/src/regression/util/init.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ const supportedSourcegraphVersionConstraint = '>=3.8'
...
-    if (version !== 'dev' && !semver.satisfies(version, supportedSourcegraphVersionConstraint)) {
+    if (version !== 'dev' && !semver.satisfies(semver.coerce(version)!, supportedSourcegraphVersionConstraint)) {
```
Comment (beyang) [inline web/src/regression/util/init.ts]:
```suggestion
    if (version !== 'dev' && !semver.satisfies(version, supportedSourcegraphVersionConstraint, { includePrerelease: true })) {
```

**Pull request merged**